[WIP] Add execute feature

jrchatruc opened the pull request:
<!-- Thank you for submitting the PR! We appreciate you spending the time to work on these changes! -->

## Motivation

This PR contains the functionality to execute program functions, needed by the CLI for the `aleo execute` feature.

Still WIP, for now it's mostly a port of the `run` code; a lot of the code can probably be abstracted as it will be shared between `run` and `execute`.

## Test Plan

<!--
    If you changed any code, please provide us with clear instructions on how you verified your changes work.
    Bonus points for screenshots and videos!
-->

## Related PRs

<!--
    If this PR adds or changes functionality, please take some time to
    update the docs at https://github.com/AleoHQ/snarkVM, and link to your PR here.
-->

jrchatruc committed:
[WIP] Add execute feature

facundoolano commented on vm/package/execute.rs:
```diff
@@ -0,0 +1,219 @@
+    }
+}
+
+impl<N: Network> Serialize for ExecutionResponse<N> {
```
not necessary to solve this in this PR, but I check if we can make this work with a derive or something like that, I'd expect not to have to write custom serializers for simple structs like this one

Commit:
Add TODO

Executing before building will fail. We need to handle the build request in snarkOS's server.

Commit:
Merge branch 'testnet3' of github.com:AleoHQ/snarkVM into feature-execute

Commit:
Revert "Merge branch 'testnet3' of github.com:AleoHQ/snarkVM into feature-execute"

This brings unwanted breaking changes for development

This reverts commit 7a3e2d3cd194d8b86670b0136f20d4b6349aae46, reversing
changes made to 1ae2d6cf24e0c69bb1f8b5334f571a493c6a6b9d.

Commit:
Handle program execution output

Commit:
Update build endpoint
